proxy/proxy/client.go

```diff
@@ -293,15 +296,36 @@ func (c *Client) Dial(instance string) (net.Conn, error) {
 	if err != nil {
 		return nil, err
 	}
-	return c.tryConnect(addr, cfg)
+	return c.tryConnect(ctx, addr, cfg)
 }
 
-func (c *Client) tryConnect(addr string, cfg *tls.Config) (net.Conn, error) {
-	d := c.Dialer
-	if d == nil {
-		d = proxy.FromEnvironment().Dial
+// Dial does the same as DialContext but using context.Background() as the context.
```

Comment:
We should consider adding a "Deprecated" notice here, since people should ideally be plumbing contexts correctly these days.

Same thought for adding a deprecated message to all of the existing things which are now duplicated into context and non-context version. If you ask me, people who still aren't plumbing contexts (especially for i/o) are just asking for tricky bugs down the line :)  (which is why I think this PR is great, thanks again for sending it out.)

Reply:
I considered adding that, but since this repo is using this for Postgres driver I thought it would be weird to say this is deprecated.

I can add the message if you want you to.

Reply:
I'm not particularly concerned about this. We can add it in the future if it becomes a serious issue.